eqcorrscan/utils/pre_processing.py

```diff
@@ -846,10 +846,39 @@ def _prep_data_for_correlation(stream, templates, template_names=None,
     # Remove templates with no matching channels
     filt = np.ones(len(template_names)).astype(bool)
     for i, template in enumerate(templates):
-        template_ids = {tr.id for tr in template}
-        if len(template_ids.intersection(stream_ids)) == 0:
+        trace_ids = {tr.id for tr in template}
+        if len(trace_ids.intersection(stream_ids)) == 0:
```

Comment:
It might be a good idea/necessary to apply this filter before the addition of empty channels for early data. This `filt` should be used to remove templates that have nothing in common with the stream, which we still want to do regardless of this issue. Removing these before adding empty channels will at best make this faster (fewer templates to check), and potentially get around any bugs whereby nan channels are added to the stream for these templates that have nothing in common with the stream.

Reply:
True; I moved the filtering to before any NaN-traces are added to the stream, and made sure all tests are carried out on the filtered set of templates.